src/java.desktop/macosx/classes/com/apple/laf/AquaTabbedPaneContrastUI.java

```diff
@@ -77,7 +80,7 @@ protected void paintTitle(final Graphics2D g2d, final Font font, final FontMetri
     protected static Color getSelectedTabTitleColor(boolean enabled, boolean pressed) {
         if (enabled && pressed) {
             return UIManager.getColor("TabbedPane.selectedTabTitlePressedColor");
-        } else if (!enabled) {
+        } else if (!enabled || (!JRSUIUtils.isMacOSXBigSurOrAbove() && !isFrameActive)) {
```

Comment:
Do we enter this condition in BigSur when SwingSet2 is not focused?

If yes, Will we not see difference in color between "new ColorUIResource(new Color(1, 1, 1, 0.55f))" and updated "black" color ?

If no, In BigSur from where black color is picked for selectedTabTitleDisabledColor?

Reply:
In BigSur, there is no difference between if frame is focused or non-focused. It is always black on white (BigSur tabbedpane text design is different than Catalina).......it's not selectedTabTitleDisabledColor which governs this but selectedTabTitleNormalColor...There the text is governed by selectedControlTextColor via this code...

`"TabbedPane.selectedTabTitleNormalColor", JRSUIUtils.isMacOSXBigSurOrAbove() ? selectedControlTextColor : selectedTabTitleNormalColor,`

Reply:
Does inactive frame lead to "enabled" having "false" value? What is the difference between "enabled" flag and "frameActive" ?

Reply:
inactive frame is non-focused frame...
enabled false is tabbedpane is not enabled...

Reply:
Let me elaborate my question, it is still not clear to me.

If i click on tabbedpane(enabled -> true), then move focus out of this Frame(SwingSet2) will that make enabled-> false? Is there any hierarchical relation between enabled & frameActive?

Reply:
No, it will not. To make enabled false, you need to call tabPane.setEnabled(false);
frameActive is to decide if frame is focused or not.

Reply:
Thanks for the clarification